tasks/managed/filter-already-released-images/tests/test-filter-large-snapshot.yaml

```diff
@@ -0,0 +1,247 @@
+    - name: dataDir
+      description: The location where data will be stored
+      type: string
+  workspaces:
+    - name: tests-workspace
+  tasks:
+    - name: setup
+      taskSpec:
+        results:
+          - name: sourceDataArtifact
+            type: string
+        volumes:
+          - name: workdir
+            emptyDir: {}
+        stepTemplate:
+          volumeMounts:
+            - mountPath: /var/workdir
+              name: workdir
+          env:
+            - name: IMAGE_EXPIRES_AFTER
+              value: $(params.ociArtifactExpiresAfter)
+            - name: ORAS_OPTIONS
+              value: $(params.orasOptions)
+            - name: DEBUG
+              value: $(params.trustedArtifactsDebug)
+        steps:
+          - name: setup-values
+            image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+            script: |
+              #!/usr/bin/env bash
+              set -eux
+
+              mkdir -p "$(params.dataDir)/$(context.pipelineRun.uid)"
+
+              # Generate a large snapshot with 150 components
+              # Mix of released and unreleased components
+              TOTAL_COMPONENTS=150
+
+              echo "Generating large snapshot with ${TOTAL_COMPONENTS} components..."
+
+              cat > "$(params.dataDir)/$(context.pipelineRun.uid)/snapshot_spec.json" << 'EOF_START'
+              {
+                "application": "large-app",
+                "components": [
+              EOF_START
+
+              for i in $(seq 0 $((TOTAL_COMPONENTS - 1))); do
+                # Determine if already released based on index
+                # Pattern: every 3rd component is released (i.e., 50 released, 100 unreleased)
+                if [ $((i % 3)) -eq 0 ]; then
+                  DIGEST="abcdefg"
+                else
+                  DIGEST="notfound"
+                fi
+
+                if [ "$i" -gt 0 ]; then
+                  echo "," >> "$(params.dataDir)/$(context.pipelineRun.uid)/snapshot_spec.json"
+                fi
+
+                # Add component
+                cat >> "$(params.dataDir)/$(context.pipelineRun.uid)/snapshot_spec.json" << EOF_COMP
+                  {
+                    "name": "component-${i}",
+                    "containerImage": "registry.io/image-${i}@sha256:${DIGEST}",
+                    "repositories": [
+                      {
+                        "url": "reg.io/target-${i}",
+                        "tags": ["v1.0", "latest"]
+                      }
+                    ]
+                  }
+              EOF_COMP
+              done
+
+              cat >> "$(params.dataDir)/$(context.pipelineRun.uid)/snapshot_spec.json" << 'EOF_END'
+                ]
+              }
+              EOF_END
+
+              echo "Generated snapshot with ${TOTAL_COMPONENTS} components"
+              echo "Pattern: Every 3rd component is 'already released'"
+              echo "Expected to filter: ~50 components"
```

Comment:
correct me if I am wrong, but the symbol`~` tells that you are expecting approximately 50 and 100 components, not exactly 50 and 100 components as your test says. This would mean that 49 would be fine or 99 it would also pass.